pkg/store/prometheus.go

```diff
@@ -339,7 +339,32 @@ func extendLset(lset []storepb.Label, extend labels.Labels) []storepb.Label {
 func (p *PrometheusStore) LabelNames(ctx context.Context, r *storepb.LabelNamesRequest) (
 	*storepb.LabelNamesResponse, error,
 ) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	u := *p.base
+	u.Path = path.Join(u.Path, "/api/v1/labels")
+
+	req, err := http.NewRequest("GET", u.String(), nil)
+	if err != nil {
+		return nil, status.Error(codes.Unknown, err.Error())
+	}
+
+	span, ctx := tracing.StartSpan(ctx, "/prom_label_names HTTP[client]")
```

Comment:
```suggestion
	span, ctx := tracing.StartSpan(ctx, "/label_names HTTP[client]")
```

Reply:
Here I followed the naming of labelValues, so should I update the value of `operationName` both `labelNames` and `labelValues` if you suggest.

Reply:
I guess you can ignore my comment then :+1:

Reply:
yea, we might want to leave this `prom_` prefix to have unique span id